### PR TITLE
Init module

### PR DESCRIPTION
**Pull request by nbucci**

### CODE REVIEW

File: values.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# https://github.com/kubernetes/ingress-nginx/blob/helm-chart-4.0.13/charts/ingress-nginx/values.yaml
+controller:
+  resources:
+    requests:
+      cpu: 250m
+      memory: 512Mi
```
Comment (paolomainardi) [inline values.yaml]: I am uncertain whether we should make these values configurable or allow module users to add other values files.

cc @Monska85

Comment (paolomainardi) [reply values.yaml]: ok we already have the additional_values attribute

Comment (paolomainardi) [inline values.yaml]: How do we calculate CPU and memory mean values?

Comment (paolomainardi) [reply values.yaml]: they look like a bit high, but my thesis is not supported by data :)

File: examples/main.tf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module "ingress_nginx" {
+  source                   = "../"
+  chart_version            = "4.8.3"
+  namespace                = "ingress-nginx"
+  host                     = "https://test"
+  cluster_ca_certificate   = base64decode("certificate")
+  token                    = "token"
+  lifecycle_ignore_changes = ["test"]
+}
```
Comment (Monska85) [inline examples/main.tf]: Please add an empty line here. To prevent this kind of issue please activate the auto-formatter.

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  version    = var.chart_version
+  namespace  = var.namespace
+
+  values = concat(
+    [templatefile(
+    "${path.module}/values.yaml",
+    {
+      ssl_reject_handshake = var.ssl_reject_handshake
+    }
+  )],
```
Comment (Monska85) [inline main.tf]: I think that we have an indentation problem here. Same as before, use the formatter to fix this issue.

File: values.yaml (hunk 1 above)
Comment (Monska85) [reply values.yaml]: We can use something _lighter_ here and eventually increase these values project by project.

File: variables.tf [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+variable "additional_values" {
+  description = "Additional values to pass to the helm chart"
+  type        = any
+  default     = {}
+}
+
+variable "ssl_reject_handshake" {
+  description = "Reject SSL handshake if it does not support SNI"
```
Comment (Monska85) [inline variables.tf]: I would prefer something like:

```suggestion
  description = "Reject SSL handshake if the client does not ask for a configured domain"
```

what do you think?

File: variables.tf [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+variable "lifecycle_ignore_changes" {
+  description = "Ignore lifecycle changes"
+  type        = list
+}
+
+variable "additional_values" {
+  description = "Additional values to pass to the helm chart"
+  type        = any
+  default     = {}
+}
```
Comment (Monska85) [inline variables.tf]: I think that the default value is an empty list (`[]`) and not an empty map (`{}`). The expected value for `values` property is a list of string. You can check the terraform documentation [here](https://registry.terraform.io/providers/hashicorp/helm/latest/docs/resources/release#values).

The `type` also could be converted into `list(string)`.

You can use [this](https://github.com/sparkfabrik/terraform-sparkfabrik-prometheus-stack/blob/main/variables.tf#L71) as an example.

**Review by Monska85 — CHANGES_REQUESTED** [review]

**Review by Monska85 — APPROVED** [review]
LGTM  :+1:

**Pull request merged**